src/Controls/src/Core/RadioButton/RadioButton.cs

```diff
@@ -223,12 +224,19 @@ public static ControlTemplate DefaultTemplate
 		{
 			get
 			{
-				if (s_defaultTemplate == null)
+				ControlTemplate defaultTemplate = null;
```

Comment:
Did you narrow down to what was leaking? I worry that fixing this via the `ControlTemplate` is just fixing a symptom instead of the cause. For example, if something in the template is leaking this will probably fix it but we probably want to fix that leak instead.

For example, if you just set the datatemplate to something really boring does it still leak?

Reply:
Yeah so it looks like part of the memory leak occurs here in the BuildDefaultTemplate method: 

```
border.SetBinding(Border.StrokeShapeProperty,
				new Binding(CornerRadiusProperty.PropertyName, converter: new CornerRadiusToShape(),
							source: RelativeBindingSource.TemplatedParent));
```